### PR TITLE
Adds postgresql persister

### PR DESCRIPTION
**Pull request by skrawcz**
This is a simple implementation that uses JSON
to serialize the state to. Very similar in spirit to the SQLLITE one, but using postgresql constructs.

I set varchar limits so they wouldn't be text.
These obviously could be tightened.

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
There's actually no real difference between varchar and text, postgres treats them the same. See https://stackoverflow.com/questions/4848964/difference-between-text-and-varchar-character-varying. 

Otherwise, looks good. Let's add a version/schema?

**Review by elijahbenizzy — COMMENTED** [review]
See comment in PR about varchar, and schema version. We should have some way that we can update it.

**Comment by skrawcz** [comment]
> See comment in PR about varchar, and schema version. We should have some way that we can update it.

I actually had it text before I changed it 😶‍🌫️ ... schema version would only make sense for the state JSON schema.

**Comment by elijahbenizzy** [comment]
Also, let's change the persister API to require `**kwargs` so we can add more of them and make it backwards compatible

File: burr/integrations/persisters/postgresql.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
+try:
```
Comment (elijahbenizzy) [inline burr/integrations/persisters/postgresql.py:1]: Let's stick this in integrations actually? Or at least use the same pattern here: https://github.com/DAGWorks-Inc/burr/blob/main/burr/tracking/client.py#L30. It automates away the message so they're all the same.

File: burr/integrations/persisters/postgresql.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    .. code:: python
+
+        p = PostgreSQLPersister.from_values("postgres", "postgres", "my_password",
+                                           "localhost", 54320, table_name="burr_state")
+
+
+    """
+
+    @classmethod
+    def from_config(cls, config: dict) -> "PostgreSQLPersister":
```
Comment (elijahbenizzy) [inline burr/integrations/persisters/postgresql.py:42]: Why do we have this? Just postgres convention?

File: burr/integrations/persisters/postgresql.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                   -v local_psql_data:/SOME/FILE_PATH/ \  # mounting a volume for data persistence
+                   -p 54320:5432 \  # port mapping
+                   -e POSTGRES_PASSWORD=my_password \  # superuser password
+                   -d postgres  # database name
+
+    Then you should be able to create the class like this:
+
+    .. code:: python
+
+        p = PostgreSQLPersister.from_values("postgres", "postgres", "my_password",
```
Comment (elijahbenizzy) [inline burr/integrations/persisters/postgresql.py:35]: persistor, not `p` -- var names

File: burr/integrations/persisters/postgresql.py (hunk 2 above)
Comment (skrawcz) [reply burr/integrations/persisters/postgresql.py:42]: I'd want it.

File: burr/integrations/persisters/postgresql.py (hunk 1 above)
Comment (skrawcz) [reply burr/integrations/persisters/postgresql.py:3]: it is in integrations, but under persisters.

**Review by elijahbenizzy — APPROVED** [review]

**Pull request merged**